### PR TITLE
Add DO examples

### PR DESCRIPTION
**Pull request by WayneDroid**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nidaqmx-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- renamed DO existing example.
- added DO examples.

### Why should this Pull Request be merged?

Add DO examples for release 1.0.

### What testing has been done?

run `poetry run pytest tests/ -q` - pass

image

### CODE REVIEW

File: examples/digital_out/cont_gen_wfm_int_clk.py
Comment (bkeryan) [inline examples/digital_out/cont_gen_wfm_int_clk.py:1]: Naming:
- cont_write_dig_port_int_clk.py or cont_gen_dig_port_int_clk.py?
- write_dig_line_int_clk.py or gen_dig_line_int_clk.py?
- write_dig_lines.py
- write_dig_port.py

Cc: @zhindes

File: examples/digital_out/update_lines.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+"""
+
+import nidaqmx
+from nidaqmx.constants import LineGrouping
+
+with nidaqmx.Task() as task:
+    data = [1, 2, 4, 8]
+
+    task.do_channels.add_do_chan(
+        "Dev1/port0/line0:3", line_grouping=LineGrouping.CHAN_FOR_ALL_LINES
```
Comment (bkeryan) [inline examples/digital_out/update_lines.py:14]: I would expect update_lines.py to use CHAN_PER_LINE like you did for DI read_lines.py

File: examples/digital_out/update_port.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+with nidaqmx.Task() as task:
+    data = 0xFFFFFFFF
+
+    task.do_channels.add_do_chan("Dev1/port0", line_grouping=LineGrouping.CHAN_FOR_ALL_LINES)
+
+    number_of_samples_written = task.write(data, auto_start=True)
+    print(f"Generate {number_of_samples_written} voltage samples.")
+    task.wait_until_done()
+    task.stop()
```
Comment (bkeryan) [inline examples/digital_out/update_port.py:18]: Number of samples and wait until done do not make sense for on-demand.

- It only generates 1 sample.
- task.write() waits until the update is complete

auto_start=True and task.stop() are still relevant though.

File: examples/digital_out/cont_gen_wfm_int_clk.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+"""Example for generating digital signal.
```
Comment (bkeryan) [inline examples/digital_out/cont_gen_wfm_int_clk.py:1]: Use an article ("generating a digital signal") or make "signal" plural ("generating digital signals").

**Review by bkeryan — CHANGES_REQUESTED** [review]

File: examples/digital_out/update_port.py (hunk 2 above)
Comment (WayneDroid) [reply examples/digital_out/update_port.py]: It's replaced to `while not task.is_task_done()` after `write()` is called.

File: examples/digital_out/cont_gen_wfm_int_clk.py (hunk 3 above)
Comment (WayneDroid) [reply examples/digital_out/cont_gen_wfm_int_clk.py]: Noted. It's updated.

File: examples/digital_out/update_port.py (hunk 2 above)
Comment (bkeryan) [reply examples/digital_out/update_port.py]: Don't spinwait. Just call stop().

According to https://www.ni.com/docs/en-US/bundle/ni-daqmx/page/whentaskdone.html an on-demand task is not done until you call stop.

File: examples/digital_out/write_dig_lines.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+with nidaqmx.Task() as task:
+    data = [True, False, True, False]
+
+    task.do_channels.add_do_chan("Dev1/port0/line0:3", line_grouping=LineGrouping.CHAN_PER_LINE)
+    task.write(data, auto_start=True)
+
+    while not task.is_task_done():
+        pass
+
+    task.stop()
```
Comment (bkeryan) [inline examples/digital_out/write_dig_lines.py:19]: Sorry for the late change in direction, but the analog output section of https://www.ni.com/docs/en-US/bundle/daqhelp/page/startfunction.html has guidance about when to use auto-start. It sounds like it's only appropriate for on-demand, and only if you want the task to auto-stop. 

Based on this, we ought to use explicit start like the C examples do.

```suggestion
    task.start()
    task.write(data)
    task.stop()
```

File: examples/digital_out/write_dig_port.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+with nidaqmx.Task() as task:
+    data = 0xFFFFFFFF
+
+    task.do_channels.add_do_chan("Dev1/port0", line_grouping=LineGrouping.CHAN_FOR_ALL_LINES)
+    task.write(data, auto_start=True)
+
+    while not task.is_task_done():
+        pass
+
+    task.stop()
```
Comment (bkeryan) [inline examples/digital_out/write_dig_port.py:19]:
```suggestion
    task.start()
    task.write(data)
    task.stop()
```

File: examples/digital_out/gen_dig_line_int_clk.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+with nidaqmx.Task() as task:
+    data: bool = [bool(i % 2) for i in range(1000)]
+
+    task.do_channels.add_do_chan("Dev1/port0/line0", line_grouping=LineGrouping.CHAN_PER_LINE)
+    task.timing.cfg_samp_clk_timing(
+        1000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=len(data)
+    )
+
+    number_of_samples_written = task.write(data, auto_start=True)
```
Comment (bkeryan) [inline examples/digital_out/gen_dig_line_int_clk.py:18]:
```suggestion
    number_of_samples_written = task.write(data)
    task.start()
```

File: examples/digital_out/gen_dig_line_int_clk.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    print(f"Generate {number_of_samples_written} voltage samples.")
```
Comment (bkeryan) [inline examples/digital_out/gen_dig_line_int_clk.py:19]:
```suggestion
    print(f"Generating {number_of_samples_written} voltage samples.")
```

**Review by bkeryan — CHANGES_REQUESTED** [review]

File: examples/digital_out/write_dig_lines.py (hunk 4 above)
Comment (WayneDroid) [reply examples/digital_out/write_dig_lines.py]: It's alright, thanks for the suggestion. It's updated as per suggestion.

**Review by bkeryan — APPROVED** [review]

**Review by zhindes — APPROVED** [review]

**Pull request merged**